### PR TITLE
Better documentation for [string_of_float].

### PR DESCRIPTION
**Pull request by fpottier**

### CODE REVIEW

**Comment by dra27** [comment]
Looks good to me - I think worth the same note on `Float.to_string` as it doesn’t actually state that it’s an alias of `Stdlib.string_of_float`

**Comment by xavierleroy** [comment]
It could also be time to reconsider the `.12` precision used in `string_of_float`.  If I remember correctly, it's because old C libraries had a tendency to add lots of unnecessary digits (e.g. 0.4000000000000001 when 0.4 would do) when higher precisions were requested.  Maybe C standard libraries have progressed since.

**Comment by fpottier** [comment]
Thanks David, I have changed the documentation of `Float.to_string` in a similar way, and documented that it is an alias for `string_of_float`.

File: stdlib/float.mli [hunk 1]
```diff
@@ -178,7 +178,12 @@ val of_string_opt: string -> float option
 (** Same as [of_string], but returns [None] instead of raising. *)
 
 val to_string : float -> string
-(** Return the string representation of a floating-point number. *)
+(** Return a string representation of a floating-point number.
+
+    This conversion can involve a loss of precision. For greater control over
+    the manner in which the number is printed, see {!Printf}.
+
+    This function is an alias for {!Stdlib.string_of_float}. *)
```
Comment (dra27) [inline stdlib/float.mli:186]: The text is fine, though apparently it has to go in `stdlib/templates/float.template.mli` as well to keep `tools/sync_stdlib_docs` happy!

Comment (fpottier) [reply stdlib/float.mli:186]: Thanks for pointing this out. Done.

**Comment by dra27** [comment]
@xavierleroy - given that we're (presumably) not going to look at changing that for 5.0, should we merge this and open an issue to track changing the precision?

**Comment by xavierleroy** [comment]
We should definitely merge this documentation update now.  But we should also run experiments with `%.17g` on our favorite platforms to see how well it behaves.

**Pull request merged**